routes/order.js

```diff
@@ -0,0 +1,129 @@
+'use strict';
+
+// Import modules
+const express = require('express');
+const Joi = require('joi');
+
+const validator = require('@byba/express-validator');
+const { checkAdmin } = require('../middleware');
+
+const {
+	helpers: {
+		SortHelper
+	}
+} = require(global.__lib);
+
+const router = express.Router();
+
+const idParamSchema = Joi.object({
+	id: Joi.string()
+		.guid()
+		.required()
+});
+
+router.get('/', checkAdmin, validator.query(Joi.object({
+	status: Joi.string().optional()
+})), async (req, res, next) => {
+	const orderWhere = {};
+	if (req.query.status && req.query.status !== 'all') {
+		orderWhere.Status = req.query.status;
+	}
+
+	const orders = (await req.db.Order.findAll({ where: orderWhere })).sort((a, b) => SortHelper.dateProp('Date', a, b, false));
+	const orderStatuses = req.db.Order.getAttributes().Status.values;
+
+	return res.render('order/index.hbs', {
+		title: 'Orders',
+		orders,
+		orderStatuses,
+		currentStatus: req.query.status ?? 'all'
+	});
+});
+
+router.get('/:id', checkAdmin, validator.params(idParamSchema), async (req, res, next) => {
```

Comment:
I'm not sure that `checkAdmin` is correct here. We need to let corps directors pay for their memberships/show entries, who will not be admin users at this point

Reply:
Agreed, but this is actually being blocked by me not working out an easy way to associate the orders/order items with an organisation or user yet.

I imagine I'll chuck it into the "custom" property of the OrderItem and maybe bubble it up and enforce that orders can only be made up of order items that belong to the same individual/organisation. Still working on this. Any ideas are welcome!

Reply:
I think this is where things unravel a bit... 

We need different checks depending on what the `OrderItem` actually links to

- `IndividualMembership`: do the `UserId`s match?

- `OrganisationMembership`: is the current user **currently** an `OrganisationContact`?

- `EventRegistration`: is the current user **currently** an `OrganisationContact`?

There is an authentication middleware somewhere (I think it might be viewing an organisation) that tries to be a little flexible, but it may not be sophisticated enough for this check...

Reply:
Let me take a proper look tomorrow when I'm not on my phone and I can compare/think through this and the models!